lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,374 @@
+    std::vector<double> &textureMin, 
+    std::vector<double> &textureMax
+) {
+    textureMin.clear();
+    textureMax.clear();
+
+    std::vector<double> boxMin = _cacheParams.boxMin;
+    std::vector<double> boxMax = _cacheParams.boxMax;
+
+    int orientation = _cacheParams.orientation;
+    if (orientation == XY) {
+        textureMin.push_back(boxMin[X]);
+        textureMin.push_back(boxMin[Y]);
+        textureMax.push_back(boxMax[X]);
+        textureMax.push_back(boxMax[Y]);
+    }
+    if (orientation == XZ) {
+        textureMin.push_back(boxMin[X]);
+        textureMin.push_back(boxMin[Z]);
+        textureMax.push_back(boxMax[X]);
+        textureMax.push_back(boxMax[Z]);
+    }
+    if (orientation == YZ) {
+        textureMin.push_back(boxMin[Y]);
+        textureMin.push_back(boxMin[Z]);
+        textureMax.push_back(boxMax[Y]);
+        textureMax.push_back(boxMax[Z]);
+    }
+}
+
+bool SliceRenderer::_isCacheDirty() const
+{
+    SliceParams* p = dynamic_cast<SliceParams*>(GetActiveParams());
+    assert(p);
+
+    if (_cacheParams.varName          != p->GetVariableName())       return true;
+    if (_cacheParams.heightVarName    != p->GetHeightVariableName()) return true;
+    if (_cacheParams.ts               != p->GetCurrentTimestep())    return true;
+    if (_cacheParams.refinementLevel  != p->GetRefinementLevel())    return true;
+    if (_cacheParams.compressionLevel != p->GetCompressionLevel())   return true;
+
+    if (_cacheParams.textureSampleRate != p->GetSampleRate())        return true;
+
+    MapperFunction *tf = p->GetMapperFunc(_cacheParams.varName);
+    vector <float> tf_lut;
+    tf->makeLut(tf_lut);
+    if (_cacheParams.tf_lut != tf_lut)                               return true;
+    if (_cacheParams.tf_minMax != tf->getMinMaxMapValue())           return true;
+
+    vector<double> min, max;
+    Box* box = p->GetBox();
+    box->GetExtents(min, max);
+    int orientation = box->GetOrientation();
+
+    if (_cacheParams.orientation      != orientation)                return true;
+    if (_cacheParams.boxMin != min)                                  return true;
+    if (_cacheParams.boxMax != max)                                  return true;
+
+    return false;
+}
+
+int SliceRenderer::_initializeGL() 
+{
+    return 0;
+}
+
+int SliceRenderer::_paintGL(bool fast) {
+    if (_isCacheDirty()) {
+        _saveCacheParams();
+    }
+
+    _initTexture();
+
+    std::vector<double> min = _cacheParams.boxMin;
+    std::vector<double> max = _cacheParams.boxMax;
+  
+    glEnable(GL_DEPTH_TEST);
+
+    LegacyGL *lgl = _glManager->legacy;
+    lgl->EnableTexture();
+
+    int orientation = _cacheParams.orientation;
+    if (orientation == XY)
+        _renderXY(min, max);
+    else if (orientation == XZ)
+        _renderXZ(min, max);
+    else if (orientation == YZ)
+        _renderYZ(min, max);
+
+    lgl->DisableTexture();
+
+    return 0;
+}
+
+void SliceRenderer::_renderXY(
```

Comment:
Couldn't renderXY, renderXZ and renderYZ be replaced with a single render() function if you simply did

max[i] := min[i]

when i is the axis corresponding to the plane normal?